runners/samza/src/main/java/org/apache/beam/runners/samza/translation/SamzaTestStreamTranslator.java

```diff
@@ -90,11 +97,54 @@ public void translate(
     ctx.registerInputMessageStream(output, inputDescriptor);
   }
 
+  @SuppressWarnings("unchecked")
   @Override
   public void translatePortable(
       PipelineNode.PTransformNode transform,
       QueryablePipeline pipeline,
       PortableTranslationContext ctx) {
-    throw new SamzaException("TestStream is not supported in portable by Samza runner");
+    final ByteString bytes = transform.getTransform().getSpec().getPayload();
+    final Coder<T> coder;
```

Comment:
Shall we wrap line 107 - 129 into a private static method, e.g. createTestStreamDecoder()? The good thing about static is that your lambda function instance won't hold on to the outer class instance.